packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java

```diff
@@ -325,27 +342,23 @@ public void writeAsStringAsync(String uriStr, String string, Map<String, Object>
     try {
       Uri uri = Uri.parse(uriStr);
       ensurePermission(uri, Permission.WRITE);
-      if ("file".equals(uri.getScheme())) {
 
-        String encoding = "utf8";
-        if (options.containsKey("encoding") && options.get("encoding") instanceof String) {
-          encoding = ((String) options.get("encoding")).toLowerCase();
-        }
+      String encoding = "utf8";
+      if (options.containsKey("encoding") && options.get("encoding") instanceof String) {
+        encoding = ((String) options.get("encoding")).toLowerCase();
+      }
 
-        FileOutputStream out = new FileOutputStream(uriToFile(uri));
+      try (OutputStream out = getOutputStream(uri)) {
         if (encoding.equals("base64")) {
           byte[] bytes = Base64.decode(string, Base64.DEFAULT);
           out.write(bytes);
         } else {
-          OutputStreamWriter writer = new OutputStreamWriter(out);
-          writer.write(string);
-          writer.close();
+          try (OutputStreamWriter writer = new OutputStreamWriter(out)) {
+            writer.write(string);
+          }
         }
-        out.close();
-        promise.resolve(null);
```

Comment:
I don't think this should be removed outright, even if it works it's much clearer to explicitly call this in the success case